File: DesignPatternsDotNet/CommandPattern/Commands/PressBack.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CommandPattern.Commands
{
   public class PressBack : ICommand{
	private IReceiver myReceiver  = null;
	public PressBack(IReceiver aReceiver ){
		myReceiver= aReceiver ;		
	}
	public void Execute() {
		myReceiver.Back();
	}
	public void Undo()// Usually back is the undo it self.
	{
		myReceiver.Back();
	}
}
}

File: DesignPatternsJava/src/com/command/PressBack.java
package com.command;

public class PressBack implements ICommand{
	private IReceiver myReceiver  = null;
	public PressBack(IReceiver aReceiver ){
		myReceiver= aReceiver ;		
	}
	public void execute() {
		myReceiver.back();
	}
	public void undo()// Usually back is the undo it self.
	{
		myReceiver.back();
	}
}
